src/identity/identity.ts

```diff
@@ -164,6 +182,46 @@ export class Identity {
 
     return blinded.toBase64();
   }
+
+  async blindPsetV2WithSource(
+    psetBase64: string,
+    lastBlinder: boolean,
+    blindingSource: {
```

Comment:
suggestion: instead of blinding source, I would pass the ZKPGenerator here. Then Identity implementation creates the generator according to the parameter passed by the user. It lets to don't touch the base class if later a new identity is using another blinding source (the ZKPGenerator complexity in the upper layer). 

Moreover "masterBlindingKey" is something related to slip77 identities only (not the case of all type of identities)

Reply:
The idea is to isolate everything related to blinding here in `blindPsetV2WithSource`. There are only 3 ways to instantiate a generator, those already supported by the method, and this ain't gonna change.

The master key is passed as buffer to not use the slip interface in other parts of the code.

The only identity using the list of blinding keys is the single-key one actually since it doesn't have a master key of course.